src/daft-sql/src/planner.rs

```diff
@@ -756,145 +755,163 @@ impl<'a> SQLPlanner<'a> {
         Ok((exprs, desc, nulls_first))
     }
 
-    fn plan_from(&mut self, from: &[TableWithJoins]) -> SQLPlannerResult<Relation> {
+    /// Plans the FROM clause of a query and populates self.current_relation and self.table_map
+    /// Should only be called once per query.
+    fn plan_from(&mut self, from: &[TableWithJoins]) -> SQLPlannerResult<()> {
         if from.len() > 1 {
             let mut from_iter = from.iter();
 
             let first = from_iter.next().unwrap();
-            let mut rel = self.new_with_context().plan_relation(&first.relation)?;
+            let mut rel = self.plan_relation(&first.relation)?;
             self.table_map.insert(rel.get_name(), rel.clone());
             for tbl in from_iter {
-                let right = self.new_with_context().plan_relation(&tbl.relation)?;
+                let right = self.plan_relation(&tbl.relation)?;
                 self.table_map.insert(right.get_name(), right.clone());
                 let right_join_prefix = Some(format!("{}.", right.get_name()));
 
                 rel.inner =
                     rel.inner
                         .cross_join(right.inner, None, right_join_prefix.as_deref())?;
             }
-            return Ok(rel);
+            self.current_relation = Some(rel);
+            return Ok(());
         }
 
         let from = from.iter().next().unwrap();
 
-        fn collect_idents(
-            left: &[Ident],
-            right: &[Ident],
-            left_rel: &Relation,
-            right_rel: &Relation,
-        ) -> SQLPlannerResult<(Vec<ExprRef>, Vec<ExprRef>)> {
-            let (left, right) = match (left, right) {
-                // both are fully qualified: `join on a.x = b.y`
-                ([tbl_a, Ident{value: col_a, ..}], [tbl_b, Ident{value: col_b, ..}]) => {
-                    if left_rel.get_name() == tbl_b.value && right_rel.get_name() == tbl_a.value {
-                        (col_b.clone(), col_a.clone())
-                    } else {
-                        (col_a.clone(), col_b.clone())
-                    }
+        macro_rules! return_non_ident_errors {
+            ($e:expr) => {
+                if !matches!(
+                    $e,
+                    PlannerError::ColumnNotFound { .. } | PlannerError::TableNotFound { .. }
+                ) {
+                    return Err($e);
                 }
-                // only one is fully qualified: `join on x = b.y`
-                ([Ident{value: col_a, ..}], [tbl_b, Ident{value: col_b, ..}]) => {
-                    if tbl_b.value == right_rel.get_name() {
-                        (col_a.clone(), col_b.clone())
-                    } else if tbl_b.value == left_rel.get_name() {
-                        (col_b.clone(), col_a.clone())
-                    } else {
-                        unsupported_sql_err!("Could not determine which table the identifiers belong to")
-                    }
+            };
```

Comment:
I use a macro here because a function would have to own the error object to return it, and I would like to use the error in later parts of the code if it is not returned.